File: src-scala/org/jgll/sppf/IntermediateNode.scala
package org.jgll.sppf

import org.jgll.grammar.slot.GrammarSlotTrait
import org.jgll.traversal.SPPFVisitorTrait
import scala.virtualization.lms.common.Base

trait IntermediateNodeTrait {
  self: GrammarSlotTrait with NonPackedNodeTrait with SPPFVisitorTrait with SPPFNodeTrait with Base =>
  class IntermediateNode(slot: GrammarSlot, leftExtent: Int, rightExtent: Rep[Int])
      extends NonPackedNode(slot, leftExtent, rightExtent) {

    override def equals(obj: Any): Boolean = {
      if (!(obj.isInstanceOf[IntermediateNode])) {
        return false
      }
      super.equals(obj)
    }

    override def accept(visitAction: SPPFVisitor) {
      visitAction.visit(this)
    }
  }
}

File: src/org/jgll/sppf/IntermediateNode.java
package org.jgll.sppf;

import org.jgll.grammar.slot.GrammarSlot;
import org.jgll.traversal.SPPFVisitor;

/**
 * 
 * @author Ali Afroozeh
 *
 */
public class IntermediateNode extends NonPackedNode {
		
	public IntermediateNode(GrammarSlot slot, int leftExtent, int rightExtent) {
		super(slot, leftExtent, rightExtent);
	}

	@Override
	public boolean equals(Object obj) {
		if(!(obj instanceof IntermediateNode)) {
			return false;
		}
		return super.equals(obj);
	}

	@Override
	public void accept(SPPFVisitor visitAction) {
		visitAction.visit(this);
	}

}